include/ezgl/canvas.hpp

```diff
@@ -35,6 +37,37 @@ namespace ezgl {
 
 class renderer;
 
+/**
+ * create_and_generate_pdf and create_and_generate_svg both create a surface and generate an file output.
+ * 
+ * @param widget        a widget which specifies the width and the height of the 
+ *                      surface generated. Can use any widget — button, window, etc. 
+ *                      Should be using the MainCanvas (GtkDrawingArea) to save the
+ *                      graphical content shown in the main canvas.
+ * @param file_name:    a filename for the PDF output (must be writable), NULL may 
+ *                      be used to specify no output. This will generate a PDF/SVG 
+ *                      surface that may be queried and used as a source, without 
+ *                      generating a temporary file.
+ * @return              a pointer to the newly created surface. The caller owns 
+ *                      the surface and should call cairo_surface_destroy() when 
+ *                      done with it. 
+ *                      Always returns a valid pointer, but it will return a pointer 
+ *                      to a NULL surface if an error such as out of memory occurs. 
+ * 
+ * sample code: http://zetcode.com/gfx/cairo/cairobackends/
+ */
```

Comment:
This is a nice comment block, but it doesn't seem to match the code.  The code seems to only use this routine to create a surface, not to create an entire file output.
If this routine created the entire file (we're done with the whole generation) why does it return a surface instead of a success / fail boolean?

Reply:
Sorry, this function doesn't explicitly generate the output file, I will make it more clear.

Reply:
Sorry, I was confused, this function does generate an output file. The reason why we need the canvas::print_pdf() function is that we need to call a callback redraw function to draw the stuffs on the newly created surface (using the returned pointer from create_and_generate_pdf), otherwise it would just be blank pdf.

Reply:
To make function names consistent with create_surface(), I would rename: create_and_generate_pdf to create_pdf_surface
create_and_generate_svg to create_svg_surface
create_png to create_png_surface

Reply:
> To make the formatting of canvas.cpp file consistent with other files in EZGL, run
> `clang-format -style=file -i canvas.cpp`
> This command uses the .clang-format file in the root directory of the repo.

I cannot run this command, I am not sure why I can't see the .clang-format file from my terminal.

Reply:
No problem, I will run it. May be clang-format is not installed on your machine.